controllers/cluster_controller_phases.go

```diff
@@ -181,10 +181,35 @@ func (r *ClusterReconciler) reconcileKubeconfig(ctx context.Context, cluster *cl
 		return nil
 	}
 
-	_, err := secret.Get(r.Client, cluster, secret.Kubeconfig)
+	machines, err := getActiveMachinesInCluster(ctx, r.Client, cluster.Namespace, cluster.Name)
+	if err != nil {
+		return errors.Wrap(err, "failed to get active machines in cluster")
+	}
+
+	foundControlPlane := false
+	for _, m := range machines {
+		if util.IsControlPlaneMachine(m) {
+			foundControlPlane = true
+			break
+		}
+	}
+
+	if !foundControlPlane {
+		// TODO: fix this after #1456 is merged
+		r.Log.Info("no controlPlane machine found, skip reconciling kuberconfig secret")
+		return nil
+	}
+
+	_, err = secret.Get(r.Client, cluster, secret.Kubeconfig)
 	switch {
 	case apierrors.IsNotFound(err):
-		return kubeconfig.CreateSecret(ctx, r.Client, cluster)
+		err = kubeconfig.CreateSecret(ctx, r.Client, cluster)
+		if apierrors.IsNotFound(err) {
```

Comment:
Instead of checking this, can we create an error in `kubeconfig` package, something like
```
ErrDependentCertificateNotFound = ....
```

and check that error here instead of using `apierrors.IsNotFound` which exposes the internal implementation of the `CreateSecret` function?